src/components/20-molecules/file-upload/index.js

```diff
@@ -208,42 +208,48 @@ class AXAFileUpload extends LitElement {
     this.requestUpdate();
   }
 
+  validateOverallSize(fileSize = 0) {
+    const {
+      sizeOfAllFilesInBytes,
+      maxSizeOfAllFilesKB,
+      filesTooBigStatusText,
+    } = this;
+
+    const maxSizeOfAllFilesInBytes = getBytesFromKilobyte(maxSizeOfAllFilesKB);
+    this.globalErrorMessage =
+      sizeOfAllFilesInBytes + fileSize > maxSizeOfAllFilesInBytes
+        ? filesTooBigStatusText
+        : '';
+  }
+
   validateFiles(compressedImages, notImagesFiles) {
-    const maxSizeOfSingleFileByte = getBytesFromKilobyte(
-      this.maxSizeOfSingleFileKB
-    );
-    const maxSizeOfAllFilesByte = getBytesFromKilobyte(
-      this.maxSizeOfAllFilesKB
+    const { maxSizeOfSingleFileKB, maxNumberOfFiles } = this;
+    const maxSizeOfSingleFileInBytes = getBytesFromKilobyte(
+      maxSizeOfSingleFileKB
     );
 
     const faultyFiles = [];
-    let finalFiles = notImagesFiles;
-
-    for (let i = 0; i < compressedImages.length; i++) {
-      if (
-        this.sizeOfAllFilesByte + compressedImages[i].size >
-        maxSizeOfAllFilesByte
-      ) {
-        this.globalErrorMessage = this.filesTooBigStatusText;
-      } else if (compressedImages[i].size > maxSizeOfSingleFileByte) {
-        faultyFiles.push(compressedImages[i]);
+    const finalFiles = [];
+    const files = [...notImagesFiles].concat(compressedImages);
+
+    for (let i = 0, n = files.length, file, fileSize; i < n; i++) {
```

Comment:
@markus-walther using the for loop var declaration for in scope not loop relevant var declaration is in my opinion not the cleanest way. Better declare on one line below the vars file and fileSize. Is more readable